File: THCS/2022_2023/YenBai_2223/Cau2/BNGUOC.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

// Hàm để đảo ngược một chuỗi
string reverseString(const string& str) {
    return string(str.rbegin(), str.rend());
}

// Hàm để so sánh số đảo ngược của A và B
string compareReversedNumbers(const string& A, const string& B) {
    string reversedA = reverseString(A);
    string reversedB = reverseString(B);
    
    // So sánh số đảo ngược
    if (reversedA > reversedB) {
        return A; // Trả về A nếu số đảo ngược của A lớn hơn
    } else {
        return B; // Trả về B nếu số đảo ngược của B lớn hơn
    }
}

int main() {
    // Đọc dữ liệu từ tệp
    ifstream infile("BNGUOC.INP");
    string A, B;
    infile >> A >> B;
    infile.close();

    // So sánh và lấy kết quả
    string result = compareReversedNumbers(A, B);

    // Ghi kết quả vào tệp
    ofstream outfile("BNGUOC.OUT");
    outfile << result << endl;
    outfile.close();

    return 0;
}

File: THCS/2022_2023/YenBai_2223/Cau2/BNGUOC.py
def reverse_number(num_str):
    """Đảo ngược chuỗi số."""
    return num_str[::-1]

def compare_reversed_numbers(A, B):
    """So sánh số đảo ngược của A và B."""
    reversed_A = reverse_number(A)
    reversed_B = reverse_number(B)
    
    # So sánh số đảo ngược
    if reversed_A > reversed_B:
        return A  # Trả về A nếu số đảo ngược của A lớn hơn
    else:
        return B  # Trả về B nếu số đảo ngược của B lớn hơn

# Đọc dữ liệu từ tệp
with open("BNGUOC.INP", "r") as infile:
    A, B = infile.read().strip().split()

# So sánh và lấy kết quả
result = compare_reversed_numbers(A, B)

# Ghi kết quả vào tệp
with open("BNGUOC.OUT", "w") as outfile:
    outfile.write(result + "\n")
